insights/client/collection_rules.py

```diff
@@ -6,15 +6,14 @@
 import json
 import logging
 import six
-import shlex
 import os
 import requests
 import yaml
 import stat
 from six.moves import configparser as ConfigParser
 
-from subprocess import Popen, PIPE, STDOUT
 from tempfile import NamedTemporaryFile
+from . import crypto
```

Comment:
Well, this allows you to mock the function more easily. If we imported the function itself, Python would create a new object instead of a reference, forcing us to track down how it is implemented. This way there is only one source for the existence of the function implementation.
Plus including the module name makes it easier for the reader to understand where the function is coming from and any additional context it may have. You don't get that from a bare function.

Overall, that is subjective, we may not find an agreement on this detail.